rust/lance-linalg/build.rs

```diff
@@ -76,8 +76,9 @@ fn main() -> Result<(), String> {
 fn build_f16_with_flags(suffix: &str, flags: &[&str]) -> Result<(), cc::Error> {
     let mut builder = cc::Build::new();
     builder
-        // TODO: why specify the compiler?
-        // .compiler("clang")
+        // We use clang #pragma to yields better vectorization
+        // See https://github.com/lancedb/lance/pull/2885
+        .compiler("clang")
```

Comment:
We can recommend `clang`, but it seems wrong to force users to use it. Newer versions of GCC should work okay (even if they aren't the fastest), and I think we should let users do that if they want.

Reply:
Well, this is a 3-5X performance difference across the board, see #2885, so I don't think this is the point where people should pick a compiler.